### ISSUE TITLE
Add in instance_type to Launch Template as ASG StartInstanceRefreshes API fails due to instance_type not set

### ISSUE THREAD

**Issue by andypaterson40** [terraform-aws-modules/terraform-aws-eks#2506]
I have noticed that instance_type is commented out here:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/eks-managed-node-group/main.tf#L174

Any reason why this is commented out? 

I'm using Karpenter and need to recycle the managed node group nodes once i have updated the AMI in the Launch Template. However, the AutoScaling API call start_instance_refresh fails as the instance_type is not set. Unfortunately unable to change this in the ec2.modify_launch_template. 

why is the above line commented out? any chance you can uncomment it for future release?

**Comment by bryantbiggs** [comment]
Did you miss the comment above the line you linked? `# Set on node group instead`?

**Comment by andypaterson40** [comment]
Yeah i saw that but that doesn't set it on the LT. So when you come to do ASG.start_instance_refreshes - it fails as instance_type is not set.

**Comment by bryantbiggs** [comment]
are you familiar with how EKS managed nodegroups work?

**Comment by andypaterson40** [comment]
Well yeah, it handles the management and lifecycle of the nodes for you and it is ultimately using an ASG. I'm trying to recycle the nodes once a new AMI is available.  Assume i should really be using a self managed nodes in this case. But was just wondering why instance_type is not set on the LT as well in the module which i referenced above?

**Comment by bryantbiggs** [comment]
So to start, I would suggest going back and re-reading what EKS managed nodegroups provide. You are currently trying to manage a managed nodegroup as if it is self-managed. 

With managed nodegroups, there are 1-2 launch templates used. This module defaults to using a custom launch template since it provides a number of features that the majority of users want to utilize. Either route, managed node groups take in 1+ instance types which is why the launch template does not set `instance_type`, its set on the nodegroup (hence the comment). This is what allows for better support for spot instances if used (you want a diverse pool of instance types to choose from). 

When you update a custom launch template, or update the Kubernetes version of the nodegroup, the nodegroup will roll out changes automatically. You can meter the amount of nodes in flux by using the update_config setting, which is synonymous with the instance_refresh setting of autoscaling groups. Again, its a managed service over common constructs (i.e. - ASG, launch template, etc.). 

I highly recommend you revisit the EKS managed nodegroup docs because the route you are trying to use is going to cause you service disruptions
